jwst/ramp_fitting/tests/test_ramp_fit_step.py

```diff
@@ -246,3 +247,232 @@ def test_int_times2(generate_miri_reffiles, setup_inputs):
     assert cube_model is not None
 
     assert(len(cube_model.int_times) == nints)
+
+
+def one_group_suppressed(nints, suppress, setup_inputs):
+    """
+    Creates three pixel ramps.
+    The first ramp has no good groups.
+    The second ramp has one good group.
+    The third ramp has all good groups.
+
+    Sets up the models to be used by the tests for the one
+    group suppression flag.
+    """
+    # Define the data.
+    ngroups, nrows, ncols = 5, 1, 3
+    dims = nints, ngroups, nrows, ncols
+    rnoise, gain = 10, 1
+    group_time, frame_time = 5.0, 1
+    rampmodel, gdq, rnModel, pixdq, err, gmodel = setup_inputs(
+        ngroups=ngroups, readnoise=rnoise, nints=nints, nrows=nrows,
+        ncols=ncols, gain=gain, deltatime=group_time)
+
+    rampmodel.meta.exposure.frame_time = frame_time
+
+    # Setup the ramp data and DQ.
+    arr = np.array([k + 1 for k in range(ngroups)], dtype=float)
+    sat = dqflags.pixel["SATURATED"]
+    sat_dq = np.array([sat] * ngroups, dtype=rampmodel.groupdq.dtype)
+    zdq = np.array([0] * ngroups, dtype=rampmodel.groupdq.dtype)
+
+    rampmodel.data[0, :, 0, 0] = arr
+    rampmodel.data[0, :, 0, 1] = arr
+    rampmodel.data[0, :, 0, 2] = arr
+
+    rampmodel.groupdq[0, :, 0, 0] = sat_dq  # All groups sat
+    rampmodel.groupdq[0, :, 0, 1] = sat_dq  # 0th good, all others sat
+    rampmodel.groupdq[0, 0, 0, 1] = 0
+    rampmodel.groupdq[0, :, 0, 2] = zdq     # All groups good
+
+    print("-" * 80)
+    print(f"rampmodel.groupdq = \n{rampmodel.groupdq}")
+    print("-" * 80)
+
+    if nints > 1:
+        rampmodel.data[1, :, 0, 0] = arr
+        rampmodel.data[1, :, 0, 1] = arr
+        rampmodel.data[1, :, 0, 2] = arr
+
+        # All good ramps
+        rampmodel.groupdq[1, :, 0, 0] = zdq
+        rampmodel.groupdq[1, :, 0, 1] = zdq
+        rampmodel.groupdq[1, :, 0, 2] = zdq
+
+    rampmodel.suppress_one_group_ramps = suppress
+
+    # Call ramp fit through the step class
+    slopes, cube_model = RampFitStep.call(
+        rampmodel,
+        override_gain=gmodel,
+        override_readnoise=rnModel,
+        suppress_one_group=suppress,
+        maximum_cores="none")
+
+    return slopes, cube_model, dims
+
+
+def test_one_group_not_suppressed_one_integration(setup_inputs):
+    slopes, cube, dims = one_group_suppressed(1, False, setup_inputs)
```

Comment:
Include short description of test purpose (also for later tests)

Reply:
I included short descriptions for the tests.